uamqp/aio/_client_async.py

```diff
@@ -209,6 +287,176 @@ async def do_work_async(self, **kwargs):
             return True
         return await self._client_run_async(**kwargs)
 
+    async def mgmt_request_async(self, message, **kwargs):
+        """
+        :param message: The message to send in the management request.
+        :type message: ~uamqp.message.Message
+        :keyword str operation: The type of operation to be performed. This value will
+         be service-specific, but common values include READ, CREATE and UPDATE.
+         This value will be added as an application property on the message.
+        :keyword str operation_type: The type on which to carry out the operation. This will
+         be specific to the entities of the service. This value will be added as
+         an application property on the message.
+        :keyword str node: The target node. Default node is `$management`.
+        :keyword float timeout: Provide an optional timeout in seconds within which a response
+         to the management request must be received.
+        :rtype: ~uamqp.message.Message
+        """
+
+        # The method also takes "status_code_field" and "status_description_field"
+        # keyword arguments as alternate names for the status code and description
+        # in the response body. Those two keyword arguments are used in Azure services only.
+        operation = kwargs.pop("operation", None)
+        operation_type = kwargs.pop("operation_type", None)
+        node = kwargs.pop("node", "$management")
+        timeout = kwargs.pop('timeout', 0)
+        try:
+            mgmt_link = self._mgmt_links[node]
+        except KeyError:
+
+            mgmt_link = ManagementOperation(self._session, endpoint=node, **kwargs)
+            self._mgmt_links[node] = mgmt_link
+            await mgmt_link.open()
+
+            while not await mgmt_link.ready():
+                await self._connection.listen(wait=False)
+
+        operation_type = operation_type or b'empty'
+        status, description, response = await mgmt_link.execute(
+            message,
+            operation=operation,
+            operation_type=operation_type,
+            timeout=timeout
+        )
+        return response
+
+
+class SendClient(SendClientSync, AMQPClientAsync):
```

Comment:
I'd like to scrap this multiple client hierarchy.... though it doesn't need to be done now.

Reply:
I also have the same kind of concern for class inheritance -- we use inherence here while not in connection/link/session, I was tempted to align the design.

I would love to create an issue for this and resolve the naming problem (`_async` suffix) in the same issue.

Reply:
created an issue: https://github.com/Azure/azure-sdk-for-python/issues/22972